prow/jenkins/controller.go

```diff
@@ -279,17 +281,22 @@ func (c *Controller) terminateDupes(pjs []kube.ProwJob, jbs map[string]JenkinsBu
 }
 
 func syncProwJobs(
+	l *logrus.Entry,
 	syncFn syncFn,
 	maxSyncRoutines int,
 	jobs <-chan kube.ProwJob,
 	reports chan<- kube.ProwJob,
 	syncErrors chan<- error,
 	jbs map[string]JenkinsBuild,
 ) {
+	goroutines := maxSyncRoutines
+	if goroutines > len(jobs) {
+		goroutines = len(jobs)
+	}
 	wg := &sync.WaitGroup{}
-	wg.Add(maxSyncRoutines)
-	// TODO: Start len(jobs) goroutines if maxSyncRoutines is greater than len(jobs).
-	for i := 0; i < maxSyncRoutines; i++ {
+	wg.Add(goroutines)
+	l.Debugf("Firing up %d goroutines", goroutines)
+	for i := 0; i < goroutines; i++ {
 		go func(jobs <-chan kube.ProwJob) {
```

Comment:
`jobs` doesn't need to be parameterized, it will be captured properly. The same applies to plank.

Reply:
removed